text/src/autogluon/text/automm/utils.py

```diff
@@ -88,11 +89,17 @@ def infer_metrics(
         Name of evaluation metric.
     """
     if eval_metric_name is not None:
-        validation_metric_name = eval_metric_name
-        return validation_metric_name, eval_metric_name
-
-    if problem_type in [MULTICLASS, BINARY]:
+        if eval_metric_name in VALID_METRICS:
```

Comment:
We can remove the second paragraph of `infer_metrics`'s docstrings since we have solved the issues.